### DISCUSSION TITLE
Release/5.2.3

### DISCUSSION THREAD

**Discussion by Athanasius** [EDCD/EDMarketConnector#1358] [Announcement]
* We now test against, and package with, Python 3.9.9.

  **As a consequence of this we no longer support Windows 7.  
  This is due to [Python 3.9.x itself not supporting Windows 7](https://www.python.org/downloads/windows/).
  The application (both EDMarketConnector.exe and EDMC.exe) will crash on startup due to a missing DLL.**

  This should have no other impact on users or plugin developers, other than the latter now being free to use features that were introduced since the Python 3.7 series.

  Developers can check the contents of the `.python-version` file in the source (it's not distributed with the Windows installer) for the currently used version in a given branch.

---

Release 5.2.3
===

This release fixes one bug and fixes some example code.

* Odyssey changed the order of some Journal events.  This caused our logic for tracking the following to break, and thus not report them ever to Inara:

    - Ship Combat, Trade and Exploration ranks.
    - On-foot Combat and Exobiologist ranks.
    - Engineer unlocks and progress.
    - Reputations with Major Factions (Superpowers).
  
  This is now fixed and the current state of all of these will be correctly reported to Inara if you have API access for it configured.

Developers
---

* Now built using Python 3.9.9.

* Updated [PLUGINS.md](https://github.com/EDCD/EDMarketConnector/blob/main/PLUGINS.md#packaging-extra-modules) to state that we don't actually include *all* of Python's standard library.

* The [click_counter](https://github.com/EDCD/EDMarketConnector/tree/main/docs/examples/click_counter) example plugin code has been corrected to both actually work fully, and pass our linting.

<hr /><em>This discussion was created from the release <a href='https://github.com/EDCD/EDMarketConnector/releases/tag/Release/5.2.3'>Release/5.2.3</a>.</em>

**Comment by iam9ripe** [comment]
Hello Friends!
Found small bug in last release. If using steam for playing ED, your program is not connected to frontier auth page.
Link for using auth frontier page via steam is changed from auth.frontierstore.net  to user.frontierstore.net :) as result your program is not proceed. Please make the changes and fix it.

I think additional page in settings menu for make changes of auth frontier page possibility is good idea. :) 

Thank you for your patience and Good play and !

**Reply by Athanasius** [reply]
1. I've just tried this with my Steam account, and it worked fine.
    1. Ran EDMC 5.2.3 (from source, under debugger)
    2. Logged into my ED game Steam account.
    3. That caused the web browser authentication flow.
    4. At the end of that flow (select Steam, 'Sign In' on Steam site, press 'Approve' on auth.frontierstore.net) I ended up at the expected localhost URL (because I was running under the debugger, so it didn't use the custom edmc:// protocol).
    5. I redid that (after removing the requisite key using regedit.exe) with the released 5.2.3 executable, and the only difference is the browser staying on the 'Approve' page of auth.frontierstore.net after pressing the button (EDMC shows 'Authentication successful as expected, and pressing 'Update' works).
  
2. The string 'user.frontierstore.net' isn't even in the EDMC code at all, only 'auth.frontierstore.net', so whatever caused you to end up on user.frontierstore.net was from the Frontier side of things.

I'd suggest you just try this again, to see if it was a transitory issue with either Steam or frontierstore.net.

If it still doesn't work then I'd try:

1. Exit out of EDMarketConnector.exe
2. Open a cmd.exe ("Command Prompt")
3. `cd "c:\Program Files (x86)\EDMarketConnector"`
4. `EDMarketConnector.exe --forget-frontier-auth`

If it still erroneously goes to the wrong URL then whilst on user.frontierstore.net:

1. Click "Authorized Applications" on the left
2. Find "Elite Dangerous Market Connector (EDCD/Athanasius)" in the list of Names
3. Click the "Deauthorize" link for it, and try again.

If it still persists then I'm not sure what else we can do to help, but if there's any chance it's a problem from our end you'd need to [properly open a Bug Report on GitHub](https://github.com/EDCD/EDMarketConnector/issues/new?assignees=&labels=bug%2C+unconfirmed&template=bug_report.md&title=) so that we have all the necessary information.

**Comment by iam9ripe** [comment]
So, have some progress.
After on auth page i Deautorize "Elite Dangerous Market Connector (EDCD/Athanasius)"
and satrt program againe, and try make authorization, i have answer from frontier server "500 internal server error". And if reload page  can see what "Elite Dangerous Market Connector (EDCD/Athanasius)" againe is Authorazed :)

But your program still not working ;(

**Reply by Athanasius** [reply]
This very much sounds like the problem is at Frontier's end.  Contact their customer support and say you're encountering errors **from their webserver** when EDMC is trying to perform oAuth2.

**Reply by iam9ripe** [reply]
Ok i will try.
